test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -52,6 +53,26 @@ TEST(validator_test, fullfill_concept)
     EXPECT_TRUE(validator_concept<file_ext_validator>);
 }
 
+TEST(validator_test, no_file)
+{
+    filesystem::path p{"./sandbox.fasta"};
+    std::string s{"./stonebox.fasta"};
+    file_existance_validator my_validator{};
+    EXPECT_THROW(my_validator(p), parser_invalid_argument);
+    EXPECT_THROW(my_validator(s), parser_invalid_argument);
+}
+
+TEST(validator_test, file_exists)
+{
+    filesystem::path p{"./"};
+    std::string s{"./"};
+    file_existance_validator my_validator{};
+    for (auto & file : filesystem::directory_iterator(p))
+        EXPECT_NO_THROW(my_validator(file));
+    for (auto & file : filesystem::directory_iterator(s))
+          EXPECT_NO_THROW(my_validator(file));
```

Comment:
How do you know the current directory contains something?

Reply:
I assumed the current directory would be the directory the test file itself is in. Is that assumption not valid?

Reply:
I don't know, the test need not be run in the directory of the binary, it could be another test directory. You could instead use our `tmp_filename()` to create a file for the test!